engine/runtime/src/main/scala/org/enso/compiler/ModuleCache.scala

```diff
@@ -29,344 +22,44 @@ import scala.util.{Failure, Success, Using}
   *
   * @param module the module for which the caches are managed.
   */
-class ModuleCache(private val module: Module) {
-
-  // === Interface ============================================================
-
-  /** Saves the provided IR into the cache for this module.
-    *
-    * @param module the module representation to be saved
-    * @param context the language context in which saving is taking place
-    * @param useGlobalCacheLocations if true, will use global caches' location, local one otherwise
-    * @return returns the location of the cache if successful, and [[None]] if
-    *         it was unable to save
-    */
-  def save(
```

Comment:
Huge simplification. Now I see where the content of the newly created `Cache.scala` file comes from.